### DISCUSSION TITLE
Netconn/conn select how many data receive

### DISCUSSION THREAD

**Discussion by davideferrario** [MaJerle/lwesp#128] [General]
I am using the lwesp_netconn_receive function.

This function receive all data from connection, at maximum LWESP_CFG_CONN_MAX_DATA_LEN (2048) byte.

Is it possible to add a parameter to function to select how many data bytes to receive?

**Comment by MaJerle** [comment]
No. But you can always decrease `LWESP_CFG_CONN_MAX_DATA_LEN` and then you will have packets of smaller data.

**Reply by davideferrario** [reply]
Ok.Because LWESP_CFG_CONN_MAX_DATA_LEN is used in both send and receive, at least is possible to define two different for SEND and RECEIVE?

**Reply by MaJerle** [reply]
OK. But there is still no point of doing what you are doing.

**Reply by davideferrario** [reply]
The reason is that I am working on a common interface between BSD socket (using lwip), LTE modem and ESP32 AT device.

recv function of BSD socket has a parameter to select how many byte receive. If possible, if I could have a same interface between lwesp library and BSD socket API, it could perfect for my application.

**Reply by MaJerle** [reply]
recv function is based on top of netconn library. socket API holds the data if packet is larger than your requested read. It must be handled one layer above.

**Reply by davideferrario** [reply]
Ok, I am going to implement something similar to socket API based on your netconn API. I am going to save data.
You can skip my requests.
Thank you
